tools/src/org.graalvm.tools.lsp.launcher/src/org/graalvm/tools/lsp/launcher/GraalLanguageServerLauncher.java

```diff
@@ -0,0 +1,201 @@
+package org.graalvm.tools.lsp.launcher;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.ThreadFactory;
+
+import org.graalvm.launcher.AbstractLanguageLauncher;
+import org.graalvm.options.OptionCategory;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Context.Builder;
+import org.graalvm.tools.lsp.api.ContextAwareExecutor;
+import org.graalvm.tools.lsp.api.ContextAwareExecutorRegistry;
+import org.graalvm.tools.lsp.api.LanguageServerBootstrapper;
+import org.graalvm.tools.lsp.api.VirtualLanguageServerFileProvider;
+import org.graalvm.tools.lsp.filesystem.LSPFileSystem;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.Instrument;
+
+public class GraalLanguageServerLauncher extends AbstractLanguageLauncher {
+    private final ArrayList<String> lspargs = new ArrayList<>();
+
+    public static void main(String[] args) {
+        new GraalLanguageServerLauncher().launch(args);
+    }
+
+    @Override
+    protected List<String> preprocessArguments(List<String> arguments, Map<String, String> polyglotOptions) {
+        ArrayList<String> unrecognized = new ArrayList<>();
+
+        for (int i = 0; i < arguments.size(); i++) {
+            String arg = arguments.get(i);
+            if (arg.startsWith("--lsp")) {
+                lspargs.add(arg.split("--")[1]);
+            } else {
+                unrecognized.add(arg);
+            }
+        }
+
+        unrecognized.add("--polyglot");
+        return unrecognized;
+    }
+
+    @Override
+    protected void launch(Builder contextBuilder) {
+        String userDirString = System.getProperty("user.home");
+        Path userDir = null;
+        if (userDirString != null) {
+            userDir = Paths.get(userDirString);
+        }
+
+        org.graalvm.polyglot.Engine.Builder newBuilder = Engine.newBuilder();
+        newBuilder.option("lsp", "").option("lsp.Languagespecific.hacks", "true");
+        for (String arg : lspargs) {
+            if (arg.indexOf('=') != -1) {
+                newBuilder.option(arg.split("=")[0], arg.split("=")[1]);
```

Comment:
This is very inefficient. Not only it calls an expensive `split` twice, it's not necessary at all, as the index of `=` is known already. Simple substring would do the job.